### PR TITLE
[PATCH] Fix complex itemstack remapping.

### PR DESCRIPTION
**Pull request by 7kasper**
Kind of band-aid, but this fixes issues such as #900 #905 with minimal changes.

### CODE REVIEW

**Comment by BBoyJD10** [comment]
Can this be merged really need this for the server!

File: src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java [hunk 1]
```diff
@@ -46,7 +45,7 @@ public static int getDataFromLegacyCombinedId(int legacyId) {
 	}
 
 	public static int getModernIdByLegacyIdData(int legacyId, int data) {
-		return fromLegacyId.get(formLegacyCombinedId(legacyId, data));
+		return fromLegacyId.getOrDefault(formLegacyCombinedId(legacyId, data), fromLegacyId.getOrDefault(formLegacyCombinedId(legacyId, 0), combinedLegacyStoneId));
```
Comment (Shevchik) [inline src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java]: Remove this hack.

File: src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java [hunk 2]
```diff
@@ -24,9 +25,7 @@ protected static void register(String modernKey, int legacyMainId, int legacyDat
 		fromLegacyId.put(combinedLegacyId, modernId);
 	}
 	static {
-		int combinedLegacyStoneId = formLegacyCombinedId(1, 0);
 		Arrays.fill(toLegacyId, combinedLegacyStoneId);
-		fromLegacyId.defaultReturnValue(1);
```
Comment (Shevchik) [inline src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java]: 1 should be replaced with combinedLegacyStoneId

File: src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java (hunk 1 above)
Comment (7kasper) [reply src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java]: How then should we get the modern id of damaged items etc?

Comment (7kasper) [reply src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java]: Ok.. but how can we know which material the item is then? Or shall we just remove the legacydata and make complex remappers for wool etc?

Comment (Shevchik) [reply src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java]: Hmm, that's true, the complex remapper won't be able to deal with that.
I guess we will just leave it like you suggested for now, and will think about better system later.

**Pull request merged**